### PR TITLE
Filesystem crawler feature

### PR DESCRIPTION
**Pull request by Afollet**
Feature for that allows internal use of Validator on the Updog filesystem. Includes refactoring and cleaning of code base.

* Adds command line client
* Adds FileReader class
* Clarifies classes in tableutilites folder
* Printy print to json output and names validator reports
* Removes empty rows in tables
* Cleans (some) SonarQube recommendations 
* Formats code to Google standard

### CODE REVIEW

File: src/main/java/org/pdxfinder/validator/tableutilities/FileReader.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    } catch (IOException e) {
+      log.error("There was an error reading the files", e);
+    }
+    return tables;
+  }
+
+  public static Table readTsvOrReturnEmpty(File file) {
+    Table dataTable = Table.create();
+    log.trace("Reading tsv file {}", file);
+    System.out.print(String.format("Reading tsv file %s\r", file));
```
Comment (mauroz77) [inline src/main/java/org/pdxfinder/validator/tableutilities/FileReader.java:99]: Do you need that System.out.print ? Same for other places where that print is used

**Review by mauroz77 — APPROVED** [review]

**Review by CsabaHalmagyi — APPROVED** [review]

File: src/test/java/org/pdxfinder/validator/tablevalidation/ValidationServiceTest.java [hunk 2]
```diff
@@ -38,23 +32,22 @@
...
+  private ValidationService validationService;
 
   @Before
   public void setUp() {
-    MockitoAnnotations.initMocks(this);
+    this.validationService = new ValidationService();
   }
 
-  @Test
+  @Ignore
```
[the remark this answers is not in the document]
Comment (Afollet) [reply src/test/java/org/pdxfinder/validator/tablevalidation/ValidationServiceTest.java:44]: Yes, I delete them. I'm not sure the best strategy to test the ValidationService. If you have any thoughts that would be great.

File: src/main/java/org/pdxfinder/validator/tableutilities/FileReader.java (hunk 1 above)
Comment (Afollet) [reply src/main/java/org/pdxfinder/validator/tableutilities/FileReader.java:99]: Yes, I believe this is deleted in later commits

Comment (Afollet) [reply src/main/java/org/pdxfinder/validator/tableutilities/FileReader.java:99]: I guess this should have used a fixup! here

**Pull request merged**